### ISSUE TITLE
Validation layer returns function pointers from incorrect Get*ProcAddr functions

### ISSUE THREAD

**Issue by gnl21** [KhronosGroup/Vulkan-ValidationLayers#462]
When running with the validation layer enabled the CTS test dEQP-VK.api.version_check.entry_points fails. This seems to be because instance function pointers are being returned from GetDeviceProcAddr, which should return NULL. Specifically, I'm seeing a non-NULL pointer being returned for the command "vkDestroyInstance" by GetDeviceProcAddr in threading.cpp.

**Comment by jzulauf-lunarg** [comment]
Triage notes: this is caused by layers using the same lookup table `name_to_funcptr_map` for both `GetDeviceProcAddr` and `GetInstanceProcAddr`.  Either this table needs to be split or tagged s.t. GDPA returns null for non device-level proc names (per Table 2. in the spec).

The table is codegen'd in 
```./object_tracker_generator.py
./parameter_validation_generator.py
./threading_generator.py
./unique_objects_generator.py
```

And hand-coded in `core_validation.cpp`

**Comment by mark-lunarg** [comment]
Fixed by PR #997.
